### PR TITLE
refactor!: remove blob sign/verify for v1.3.0-rc.1 release

### PR DESCRIPTION
**Pull request by Two-Hearts**
This PR is for `release-1.3` branch.

It cherry picks commit from v1.2 to remove blob sign/verify for v1.3.0-rc.1 release. 

For easier review, please see https://github.com/notaryproject/notation/compare/v1.2.0..ec32bef8a58dce5447e8be43f655fe457b8272ad for your reference.

### CODE REVIEW

**Comment by Two-Hearts** [comment]
This PR will be updated after dependent PR https://github.com/notaryproject/notation-go/pull/464 is merged.

**Comment by Two-Hearts** [comment]
For easier review, please see https://github.com/notaryproject/notation/compare/v1.2.0..963df7ecc7bebab4b365d844fb5b163cd6cae677 for your reference.

**Review by shizhMSFT — DISMISSED** [review]
LGTM

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ go 1.23
 
 require (
 	github.com/notaryproject/notation-core-go v1.2.0-rc.1
-	github.com/notaryproject/notation-go v1.2.0-beta.1.0.20240926015724-84c2ec076201
+	github.com/notaryproject/notation-go v1.2.0-beta.1.0.20240927002524-3c4948808154
```
Comment (shizhMSFT) [inline go.mod:7]: We need to update the `notation-go` when `v1.3.0-rc.1` is released.

Comment (Two-Hearts) [reply go.mod:7]: Correct. We are 1 vote away from the release at the moment.

Comment (Two-Hearts) [reply go.mod]: Updated.

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**